### PR TITLE
FIX: issue #3223 (make the brace string `mold` like rebol)

### PR DESCRIPTION
**Pull request by bitbegin**

### CODE REVIEW

**Comment by bitbegin** [comment]
i have fixed #3223 , and left another issue waiting to fix.

In Red:
```
>> a: "a{}bc"
== {a{}bc}
>> 
```

In Rebol:
```
>> a: "a{}bc"
== "a{}bc"
>>
```

**Comment by bitbegin** [comment]
```
>> a: "{{{}"
== "{{{}"
>> mold a
== {"^{^{^{^}"}
>> a: "{}{"
== "{}{"
>> mold a
== {"^{^}^{"}
>> a: "{}{}}"
== "{}{}}"
>> mold a
== {"{}{}^}"}
>>
```

**Comment by bitbegin** [comment]
some difference with Rebol:
In rebol:
```
== "{}{"
>> mold a
== {"^{^}^{"}
>> a: "}{}"
== "}{}"
>> mold a
== {"^}{}"}
>>
```
fixed in Red:

```
>> a: "{}{"
== "{}{"
>> mold a
== {"{}^{"}
>> a: "}{}"
== "}{}"
>> mold a
== {"^}{}"}
>> 
```

`a: {^{}` will cause syntax error in Rebol, fixed in Red:

```
>> a: {^{}
== "{"
>> mold a
== {"^{"}
>> 
```

File: tests/source/units/all-tests.txt [hunk 1]
```diff
@@ -21,6 +21,7 @@ append-test.red
 insert-test.red
 move-test.red
 make-test.red
+mold.red
```
Comment (greggirwin) [inline tests/source/units/all-tests.txt]: Suggest adding "-test" to the filename to match others.

**Comment by greggirwin** [comment]
Very nice test suite.

**Pull request merged**